cord19-explorer/src/main/java/edu/asu/diging/cord19/explorer/core/service/worker/impl/CoordinateCleanerImpl.java

```diff
@@ -0,0 +1,246 @@
+                }
+                pubRepo.save(pub);   
+            }
+        }
+        
+        calculateCountryStats();
+        
+        task.setStatus(TaskStatus.DONE);
+        taskRepo.save((TaskImpl) task);
+    }
+    
+    
+    public Publication cleanCoordinatesImport(Publication pub) {
+        List<PersonImpl> authors = pub.getMetadata().getAuthors();
+        for(PersonImpl author : authors) {
+            List<String> coords = new ArrayList<String>();
+            List<Double> formattedCoords = new ArrayList<Double>();
+            if(author.getAffiliation().getSelectedWikiarticle() != null) {
+               WikipediaArticleImpl article = author.getAffiliation().getSelectedWikiarticle();
+               String [] splitted =  article.getCoordinates().split("\\|");
+               // split on pipes then use Ordinals to gather Degree, Mins, Secs then compute Decimal Degrees
+               //Refactor into function
+               for (String s: splitted) {      
+                   if(s.matches("-?\\d+(\\.\\d+)?")) {
+                       coords.add(s);
+                       
+                   } else if(s.equals("N") || s.equals("E")){
+                       if(coords.size() == 3) {
+                           formattedCoords.add(processLength3Coords(coords));
+                           coords.clear();
+                       }
+                       else if(coords.size() == 2) {
+                           formattedCoords.add(processLength2Coords(coords));
+                           coords.clear();
+                       }
+                       else if(coords.size() == 1) {
+                           Double doubleCoord = Double.parseDouble(coords.get(0));
+                           formattedCoords.add(doubleCoord);
+                           coords.clear();
+                       }
+                      
+                   } else if(s.equals("S") || s.equals("W")) {
+                       // handle negative coords
+                       if(coords.size() == 3) {
+                           formattedCoords.add(processLength3Coords(coords) * -1);
+                           coords.clear();
+                       }
+                       else if(coords.size() == 2) {
+                           formattedCoords.add(processLength2Coords(coords) * -1);
+                           coords.clear();
+                       }
+                       else if(coords.size() == 1) {                
+                           Double doubleCoord = Double.parseDouble(coords.get(0)) * -1;
+                           formattedCoords.add(doubleCoord);
+                           coords.clear();
+                       }
+                   }
+              }
+               CleanedCoordinatesImpl cleanedCoords = new CleanedCoordinatesImpl();
+               cleanedCoords.setType(CoordType.Point);
+               cleanedCoords.setCoordinates(formattedCoords);
+               article.setCleanedCoords(cleanedCoords);
```

Comment:
what's the different between this method and `parseCoords`? looks like a lot of duplicate code.